helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java

```diff
@@ -174,7 +174,10 @@ public void testP2PStateTransitionEnabled() {
       verifyP2PEnabled(startTime);
     }
 
-    Assert.assertEquals(p2pTrigged, total);
+    // The success rate really depends on how quick participant act in relationship with controller.
+    // For now, we set 90% threshold.
+    long threshold = Math.round(total * 0.9);
```

Comment:
@zhangmeng916 and @lei-xia , could you please confirm if this is the expectation? I understand that Kai needs to change it since the current run does not always succeed with 100% threshold. But I'm not sure if this should be our expectation?

Reply:
@jiajunwang, I sync-ed with Meng offline. @zhangmeng916, can you help to comment?

Reply:
I think we cannot guarantee 100% P2P for all kinds of infrastructure. Although local tests always passed, there're some failures in our testing environment. The percent is always above 90% though. I believe there are corner cases that we cannot label all stale messages, which will block the sending of P2P.

Reply:
Please make sure other reviewers are ok with this too.